tests/test_petsc.py

```diff
@@ -319,3 +320,45 @@ def test_cinterface_petsc_struct():
     assert any(isinstance(i, PETScStruct) for i in op.parameters)
     assert 'struct MatContext\n{' not in ccode
     assert 'struct MatContext\n{' in hcode
+
+
+@skipif('petsc')
+def test_separate_eqn():
```

Comment:
Might be worth parametrising this to try a bunch of equations including non-trivial ones